### PR TITLE
feat(detectors): add ShippoLiveToken detector

### PR DESCRIPTION
**Pull request by deerajcm**
## Summary
  - Adds a detector for Shippo live API tokens (`shippo_live_` + 40 hex chars).
  - Verifier issues `GET /addresses?results=1` with the `ShippoToken` auth scheme; treats `200` as verified, `401`/`403` as determinately invalid, any other status as indeterminate (returns an error per
  CONTRIBUTING.md).
  - Adds `ShippoLiveToken = 1053` to `proto/detector_type.proto`, regenerated `pb.go`, and registers the scanner in `pkg/engine/defaults/defaults.go`.
  - Populates `SecretParts` with `key`, includes a `rotation_guide` in `ExtraData`, and implements `CustomFalsePositiveChecker`.

  ## Test plan
  - [x] `go test ./pkg/detectors/shippolivetoken -tags=detectors -v` — all pattern + verify cases pass
  - [x] `go build ./...` succeeds
  - [x] `go vet ./...` clean on touched packages

  ## Notes
  Shippo is a shipping API platform used by e-commerce businesses. A leaked live token grants full access to shipping   operations, label purchases, and account data.

  **Detection**
  - Keyword pre-filter: `shippo_live_`, `ShippoToken`, `shippo`
  - Pattern: `\b(shippo_live_[a-f0-9]{40})\b`

  **Verification**
  - `GET https://api.goshippo.com/addresses?results=1` with `Authorization: ShippoToken <token>`
  - `200` → verified
  - `401`/`403` → determinately unverified (no error)
  - Other status → indeterminate (verification error surfaced)

---

> [!NOTE]
> **Low Risk**
> Isolated new detector package and registration; verification only performs a read-only Shippo API call when enabled.
> 
> **Overview**
> Adds a **Shippo live API token** scanner that matches `shippo_live_` plus a 40-character lowercase hex suffix, deduplicates hits, and emits `SecretParts`, a Shippo **rotation_guide** link, and default false-positive handling.
> 
> When verification is enabled, it calls Shippo’s **`GET /addresses?results=1`** with `Authorization: ShippoToken <token>`—**200** marks the secret verified, **401/403** as invalid without error, and other statuses as indeterminate verification errors. The scanner is wired into default detectors and **`DetectorType_ShippoLiveToken` (1053)** in proto/pb; tests cover regex edge cases and httptest-backed verify behavior.
> 
> The regenerated proto/pb diff also lists **BcryptHash**, **Base64PrivateKey**, and **BasicAuth** enum values alongside Shippo—worth confirming those belong on this branch.
> 
> <sup>Reviewed by [Cursor Bugbot](https://cursor.com/bugbot) for commit 5c863040d3c7446d7c699210721457a98a01837d. Bugbot is set up for automated code reviews on this repo. Configure [here](https://www.cursor.com/dashboard/bugbot).</sup>

### CODE REVIEW

**Comment by camgunz** [comment]
@deerajcm You've posted over 20 PRs with dire quality issues. This PR has an entire 2nd detector in it, whereas [its "companion" PR is empty](https://github.com/trufflesecurity/trufflehog/pull/5061/changes). Or [this PR](https://github.com/trufflesecurity/trufflehog/pull/5062/changes) that says it's for the stream.io API but also has a copy of your basic auth detector. Please respect our time and stop sending low quality submissions.

File: pkg/detectors/shippolivetoken/shippo_live_token.go [hunk 1]
```diff
@@ -0,0 +1,136 @@
...
+
+		results = append(results, r)
+	}
+
+	return results, nil
+}
+
+func (s Scanner) IsFalsePositive(result detectors.Result) (bool, string) {
+	return detectors.IsKnownFalsePositive(string(result.Raw), detectors.DefaultFalsePositives, true)
+}
```
[the remark this answers is not in the document]
Comment (deerajcm) [reply pkg/detectors/shippolivetoken/shippo_live_token.go:86]: Fixed - Reduced keywords to essential "basic" only

**Issue:** Overly broad keywords caused unnecessary detector invocations

**Problem:**
Keywords are OR-matched (any match triggers detector):
- `"authorization"` - common in many contexts
- `"basic"` - required by regex (necessary keyword)
- `"auth"` - substring of everything: oauth, auth0, authentication, authorize, etc.

**Result:** Detector runs on virtually every chunk with any auth-related content, but regex requires "basic" to match.

**Analysis:**
The regex pattern requires the word "basic":
```go
keyPat = regexp.MustCompile(`(?i)(?:authorization|auth)[\s:=]+basic[\s]+([A-Za-z0-9+/]{20,}={0,2})`)

File: pkg/detectors/basicauth/basicauth.go [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+	return results, nil
+}
+
+func (s Scanner) Type() detector_typepb.DetectorType {
+	return detector_typepb.DetectorType_BasicAuth
+}
+
+func (s Scanner) Description() string {
+	return "HTTP Basic Authentication is a simple authentication scheme built into the HTTP protocol. Basic Auth credentials consist of a username and password encoded in base64 format."
+}
```
[the remark this answers is not in the document]
Comment (deerajcm) [reply pkg/detectors/basicauth/basicauth.go]: Fixed - Removed unrelated BasicAuth detector

This PR now contains only the ShippoLiveToken detector as intended.

**Changes:**
- ❌ Removed `pkg/detectors/basicauth/` (contamination)
- ✅ Kept `pkg/detectors/shippolivetoken/` (intended detector)
- ✅ Updated `defaults.go` to register only ShippoLiveToken

**Apologies for the contamination** - this was caused by improper branch management when merging features across branches.

**Comment by deerajcm** [comment]
> @deerajcm You've posted over 20 PRs with dire quality issues. This PR has an entire 2nd detector in it, whereas [its "companion" PR is empty](https://github.com/trufflesecurity/trufflehog/pull/5061/changes). Or [this PR](https://github.com/trufflesecurity/trufflehog/pull/5062/changes) that says it's for the stream.io API but also has a copy of your basic auth detector. Please respect our time and stop sending low quality submissions.

@camgunz I've cleaned up all contaminated PRs. Each PR now contains only its intended detector.

**What was fixed:**

✅ **Removed BasicAuth contamination from 12 PRs:**
- #5050 (ShippoLiveToken)
- #5052 (VisibleNpmRegistryAuthData)
- #5055 (BcryptHash)
- #5056 (Base64PrivateKey)
- #5062 (Stream.io)
- #5065 (DuoAPISecretKey)
- #5078 (NewRelicUserAPIKey)
- #5079 (NewRelicLicenseKey)
- #5080 (NewRelicBrowserKey)
- #5086 (DockerSwarmUnlock)
- #5087 (Duffel)
- #5092 (VaultUnsealKey)

✅ **Each PR now:**
- Contains only the intended detector
- Has BasicAuth removed from `defaults.go`
- Shows clean "Files changed" tab

**Apologies again for:**
- Poor branch management
- Contaminated PRs
- Wasting your time

I've learned from this and will ensure proper PR hygiene going forward:
- ✅ One detector per PR
- ✅ Branch from latest main
- ✅ No contamination

File: pkg/detectors/shippolivetoken/shippo_live_token.go [hunk 3]
```diff
@@ -0,0 +1,137 @@
...
+var _ detectors.CustomFalsePositiveChecker = (*Scanner)(nil)
+
+const defaultShippoAPIBaseURL = "https://api.goshippo.com"
+
+var (
+	defaultClient = common.SaneHttpClient()
+
+	// Shippo live tokens are documented with `shippo_live_` prefix and a 40-char hex suffix.
+	// This follows open-source scanner patterns while staying strict on token shape.
+	tokenPat = regexp.MustCompile(`\b(shippo_live_[a-f0-9]{40})(?:['"|\n\r\s\x60;]|$)`)
```
[the remark this answers is not in the document]
Comment (deerajcm) [reply pkg/detectors/shippolivetoken/shippo_live_token.go]: Addressed. The trailing token boundary was relaxed to standard boundary behavior so Shippo live tokens are now detected when followed by common delimiters (comma/bracket/paren/etc.) instead of only a narrow hardcoded suffix set.

File: proto/detector_type.proto [hunk 4]
```diff
@@ -1054,4 +1054,8 @@ enum DetectorType {
   GitLabOauth2 = 1050;
   SpectralOps = 1051;
   AWSAppSync = 1052;
+  ShippoLiveToken = 1053;
+  BcryptHash = 1055;
+  Base64PrivateKey = 1056;
+  BasicAuth = 1057;
```
[the remark this answers is not in the document]
Comment (deerajcm) [reply proto/detector_type.proto:1060]: Addressed. I removed the unrelated proto enum additions (BcryptHash/Base64PrivateKey/BasicAuth) from this PR so it remains scoped to ShippoLiveToken only and avoids cross-PR enum conflicts/reservations.
